packages/SwingSet/test/util.js

```diff
@@ -99,8 +103,27 @@ export function capargsOneSlot(slot, iface = 'export') {
   return capargs([{ '@qclass': 'slot', iface, index: 0 }], [slot]);
 }
 
-export function makeMessage(target, method, args, result = null) {
-  const msg = { method, args, result };
+export function makeMessage(
+  target,
+  method,
+  args = [],
+  slots = [],
+  result = null,
+) {
+  // Note:
+  //   args === [] --> empty arg list
+  //   args === null --> no arg list
+  // Normally these are interchangeable, but for testing purposes we sometimes
+  // need to generate one and sometimes the other, because even though the
+  // message system doesn't care, the comparison done by t.deepEquals() does.
```

Comment:
as discussed, this will become false, and we should provide `args = []` even if our test doesn't care about the args (unless we're explicitly testing eventual-get, in which case args=undefined or args=null or something.. I never know how that works with default arguments in function definitions)

Hm, that reminds me, the difference between `null` and `undefined` in JSON serialized situations is significant, as well as default arguments. We use JSON serialization for the worker-to-vat pipe. Now that both `method` and `args` are in a *marshal*led capdata, we'll trivially preserve `method = undefined` (indicating eventual-send-function, vs the usual eventual-send-method), as well as `args = undefined` (indicating eventual-get). Previously, with `method` being sent separately, we needed to worry about the worker-to-vat pipe converting an eventual-send-function's `method = undefined` into `method = null`. (I suppose liveslots just looks for truthyness, so the difference was irrelevant, but I know we've had problems with transcript equality here).

I guess the place to pay attention is functions like this `makeMessage` where `method` or `args` are coming from something other than liveslots and HandledPromise handle invocations. It might not be trivial for `makeMessage` to sense someone calling it with an explicit `args = undefined` (as an attempt to trigger eventual-get).

That's probably fine, I don't think we'll use `makeMessage` to construct an eventual-get operation, and if we do we might declare that `makeMessage (args=null)` is how you say "please build me the same message that liveslots would have created for `args=undefined`".

But yeah, I think this should do `body = [method, args]` unconditionally, and never `body = [method]`. That might mean additional churn, if we have test code calling `makeMessage` with `args = null` for some reason.